content/en/community/hackathons/sessions/bincompat/content/Work-Items_00.-Setup.md

```diff
@@ -1,25 +1,89 @@
+oO   oO ' _ `| | |/ /  _)' _` | |_|  _)
+oOo oOO| | | | |   (| | | (_) |  _) :_
+ OoOoO ._, ._:_:_,\_._,  .__,_:_, \___)
+                  Atlas 0.13.1~d20aa7cb
+[...]
+brk(NULL) = va:0x400200000
+brk(va:0x400221000) = va:0x400221000
+Hello, World!
+write(fd:1, "Hello, World!\x0A", 14) = 14
+qemu-system-x86_64: terminating on signal 2
+```
+
+This will run a dynamically linked `helloworld` application.
+You will see a lot of output because, by default, the `run_app.sh` script uses a prebuilt version of `app-elfloader`, which has `strace`-like output enabled, for debugging purposes.
+You can see a list of all existing available application that you can run in binary compatibility mode by running `./run_app.sh` with no arguments.
+**Note: If you are running an application that requires networking support, you will need to run it with `sudo`.***
+
+If you want more control over the application, you can use the `run.sh` script.
+The `elfloader` application will need to be passed a filesystem that contains all the dependencies (i.e. external libraries) the already compiled application needs, along with the [system loader](https://www.man7.org/linux/man-pages/man8/ld.so.8.html), and will expect as an argument the name of the `ELF` that will be executed.
+The `dynamic-apps` repository already has those prepared, as you can see below:
+
+```console
+$ cd path/to/dynamic-apps-repo
 
-$ ./do.sh run
-'run' command requires target application as argument
-Target applications: helloworld_static server_static helloworld_go_static server_go_static helloworld_cpp_static helloworld_rust_static_musl helloworld_rust_static_gnu nginx_static redis_static sqlite3 bc_static gzip_static helloworld server helloworld_go server_go helloworld_cpp helloworld_rust nginx redis sqlite3 bc gzip
+$ ldd lang/c++/helloworld
+    linux-vdso.so.1 (0x00007ffd60841000)
+    libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fe9f27d3000)
+    libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fe9f25e1000)
+    libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fe9f2492000)
+    /lib64/ld-linux-x86-64.so.2 (0x00007fe9f2c1e000)
+    libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fe9f2477000)
 
-$ ./do.sh run helloworld
-[...]                       # many messages
+$ tree lang/c++/
+lang/c++/
+|-- helloworld
+|-- helloworld.cpp
+|-- lib
+|   `-- x86_64
+|       |-- libc.so.6
+|       |-- libgcc_s.so.1
+|       `-- libm.so.6
+|-- lib64
+|   `-- ld-linux-x86-64.so.2
+|-- Makefile
+`-- usr
+    `-- lib
+        `-- x86_64-linux-gnu
+            `-- libstdc++.so.6
 
-$ ./do.sh run sqlite3
-[...]                       # many messages
+6 directories, 8 files
+```
+
+You can see the dependencies of an application by running [`ldd`](https://www.man7.org/linux/man-pages/man1/ldd.1.html).
+As you can see, all the dependencies needed to run `apphelloworld` are located in the root `c++/` directory.
+You can ignore the `linux-vdso.so.1` for now.
```

Comment:
Mingle these explanations with the commands. Do not show explanations at the end, show them inline.